tests/model_monitoring/test_application_steps.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import logging
+import typing
+from unittest.mock import Mock, patch
+
+import pandas as pd
+import pytest
+
+import mlrun.artifacts
+import mlrun.common.schemas.model_monitoring.constants as mm_constants
+import mlrun.model_monitoring.applications.context as mm_context
+from mlrun.model_monitoring.applications import (
+    ModelMonitoringApplicationMetric,
+    ModelMonitoringApplicationResult,
+)
+from mlrun.model_monitoring.applications.application_steps import (
+    _PushToMonitoringWriter,
+)
+from mlrun.utils import Logger
+
+STREAM_PATH = "./test_stream.json"
```

Comment:
Please do not add this empty resource file. Use pytest's temp file fixture utility instead.